src/frontend/src/optimizer/plan_node/logical_iceberg_scan.rs

```diff
@@ -34,14 +37,46 @@ use crate::optimizer::plan_node::{
 use crate::utils::{ColIndexMapping, Condition};
 
 /// `LogicalIcebergScan` is only used by batch queries. At the beginning of the batch query optimization, `LogicalSource` with a iceberg property would be converted into a `LogicalIcebergScan`.
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone)]
 pub struct LogicalIcebergScan {
     pub base: PlanBase<Logical>,
     pub core: generic::Source,
     pub iceberg_scan_type: IcebergScanType,
+    pub predicate: IcebergPredicate,
```

Comment:
I think we should use `predicate: Condition` instead of `predicate: IcebergPredicate` in LogicalIcebergScan. Only convert it to IcebergPredicate when we need to convert LogicalIcebergScan to BatchIcebergScan.